src/NServiceBus.RabbitMQ/Receiving/MessageConverter.cs

```diff
@@ -33,36 +33,44 @@ public string RetrieveMessageId(BasicDeliverEventArgs message, Dictionary<string
         public Dictionary<string, string> RetrieveHeaders(BasicDeliverEventArgs message)
         {
             var properties = message.BasicProperties;
+            var headers = properties.Headers;
```

Comment:
> Does the Headers property have side effects, or does is just return the value of a field? If the latter (and what the semantics of a property getter convey) then it should be fine to call it more than once.

No, it was just an attempt to reduce method calls here, since it didn't look like the JIT was able to inline the getters when I took a look at the generated assembly from the disassembly window.

So, yeah sort of a micro-optimization, but since this is called on every incoming message, I thought it might be worth doing. I can drop it and simplify the code here if you prefer.